--- mono-tools-2.0.orig/configure.in
+++ mono-tools-2.0/configure.in
@@ -7,8 +7,8 @@
 AC_PROG_INSTALL
 
 dnl C# compiler
-AC_PATH_PROG(MCS, mcs, no)
-AC_PATH_PROG(GMCS, gmcs, no)
+AC_PATH_PROG(MCS, mono-csc, no)
+AC_PATH_PROG(GMCS, mono-csc, no)
 AC_PATH_PROG(RUNTIME, mono, no)
 AC_PATH_PROG(GACUTIL, gacutil, no)
 
--- mono-tools-2.0.orig/Mono.Profiler/profiler-file-decoder/Makefile.am
+++ mono-tools-2.0/Mono.Profiler/profiler-file-decoder/Makefile.am
@@ -4,7 +4,7 @@
 # Warning: This is an automatically generated file, do not edit!
 
 if ENABLE_DEBUG
-ASSEMBLY_COMPILER_COMMAND = gmcs
+ASSEMBLY_COMPILER_COMMAND = mono-csc
 ASSEMBLY_COMPILER_FLAGS =  -noconfig -codepage:utf8 -warn:4 -optimize+ -debug -define:DEBUG
 ASSEMBLY = bin/Debug/mprof-decoder.exe
 ASSEMBLY_MDB = $(ASSEMBLY).mdb
@@ -25,7 +25,7 @@
 endif
 
 if ENABLE_RELEASE
-ASSEMBLY_COMPILER_COMMAND = gmcs
+ASSEMBLY_COMPILER_COMMAND = mono-csc
 ASSEMBLY_COMPILER_FLAGS =  -noconfig -codepage:utf8 -warn:4 -optimize+
 ASSEMBLY = bin/Release/mprof-decoder.exe
 ASSEMBLY_MDB = 
--- mono-tools-2.0.orig/Mono.Profiler/profiler-decoder-library/Makefile.am
+++ mono-tools-2.0/Mono.Profiler/profiler-decoder-library/Makefile.am
@@ -4,7 +4,7 @@
 # Warning: This is an automatically generated file, do not edit!
 
 if ENABLE_DEBUG
-ASSEMBLY_COMPILER_COMMAND = gmcs
+ASSEMBLY_COMPILER_COMMAND = mono-csc
 ASSEMBLY_COMPILER_FLAGS =  -noconfig -codepage:utf8 -warn:4 -optimize+ -debug -define:DEBUG
 ASSEMBLY = bin/Debug/mprof-decoder-library.dll
 ASSEMBLY_MDB = $(ASSEMBLY).mdb
@@ -18,7 +18,7 @@
 endif
 
 if ENABLE_RELEASE
-ASSEMBLY_COMPILER_COMMAND = gmcs
+ASSEMBLY_COMPILER_COMMAND = mono-csc
 ASSEMBLY_COMPILER_FLAGS =  -noconfig -codepage:utf8 -warn:4 -optimize+
 ASSEMBLY = bin/Release/profiler-decoder-library.dll
 ASSEMBLY_MDB = 
--- mono-tools-2.0.orig/Mono.Profiler/heap-snapshot-explorer/Makefile.am
+++ mono-tools-2.0/Mono.Profiler/heap-snapshot-explorer/Makefile.am
@@ -4,7 +4,7 @@
 # Warning: This is an automatically generated file, do not edit!
 
 if ENABLE_DEBUG
-ASSEMBLY_COMPILER_COMMAND = gmcs
+ASSEMBLY_COMPILER_COMMAND = mono-csc
 ASSEMBLY_COMPILER_FLAGS =  -noconfig -codepage:utf8 -warn:4 -optimize+ -debug "-define:DEBUG"
 
 ASSEMBLY = bin/Debug/mprof-heap-snapshot-explorer.dll
@@ -25,7 +25,7 @@
 endif
 
 if ENABLE_RELEASE
-ASSEMBLY_COMPILER_COMMAND = gmcs
+ASSEMBLY_COMPILER_COMMAND = mono-csc
 ASSEMBLY_COMPILER_FLAGS =  -noconfig -codepage:utf8 -warn:4 -optimize+
 ASSEMBLY = bin/Release/heap-snapshot-explorer.dll
 ASSEMBLY_MDB = 
--- mono-tools-2.0.orig/Mono.Profiler/heap-snapshot-viewer/Makefile.am
+++ mono-tools-2.0/Mono.Profiler/heap-snapshot-viewer/Makefile.am
@@ -4,7 +4,7 @@
 # Warning: This is an automatically generated file, do not edit!
 
 if ENABLE_DEBUG
-ASSEMBLY_COMPILER_COMMAND = gmcs
+ASSEMBLY_COMPILER_COMMAND = mono-csc
 ASSEMBLY_COMPILER_FLAGS =  -noconfig -codepage:utf8 -warn:4 -optimize+ -debug "-define:DEBUG"
 
 ASSEMBLY = bin/Debug/mprof-heap-viewer.exe
@@ -33,7 +33,7 @@
 endif
 
 if ENABLE_RELEASE
-ASSEMBLY_COMPILER_COMMAND = gmcs
+ASSEMBLY_COMPILER_COMMAND = mono-csc
 ASSEMBLY_COMPILER_FLAGS =  -noconfig -codepage:utf8 -warn:4 -optimize+
 ASSEMBLY = bin/Release/mprof-heap-viewer.exe
 ASSEMBLY_MDB = 
