Index: mono-tools.git/configure.in
===================================================================
--- mono-tools.git.orig/configure.in	2013-10-10 10:41:50.863182158 +0200
+++ mono-tools.git/configure.in	2013-10-10 10:41:50.855181560 +0200
@@ -10,9 +10,9 @@
 AC_SUBST(ACLOCAL_FLAGS)
 
 dnl C# compiler
-AC_PATH_PROG(MCS, mcs, no)
-AC_PATH_PROG(GMCS, gmcs, no)
-AC_PATH_PROG(DMCS, dmcs, no)
+AC_PATH_PROG(MCS, mono-csc, no)
+AC_PATH_PROG(GMCS, mono-csc, no)
+AC_PATH_PROG(DMCS, mono-csc, no)
 AC_PATH_PROG(RUNTIME, mono, no)
 AC_PATH_PROG(GACUTIL, gacutil, no)
 
Index: mono-tools.git/Mono.Profiler/compile
===================================================================
--- mono-tools.git.orig/Mono.Profiler/compile	2013-10-10 10:41:50.863182158 +0200
+++ mono-tools.git/Mono.Profiler/compile	2013-10-10 10:41:50.859181859 +0200
@@ -1 +1 @@
-gmcs -debug -out:profiler-decoder.exe profiler-decoder-library/*.cs profiler-file-decoder/*.cs
+mono-csc -debug -out:profiler-decoder.exe profiler-decoder-library/*.cs profiler-file-decoder/*.cs
