Index: mono-tools/configure.in
===================================================================
--- mono-tools.orig/configure.in
+++ mono-tools/configure.in
@@ -10,9 +10,9 @@ AC_PROG_INSTALL
 AC_SUBST(ACLOCAL_FLAGS)
 
 dnl C# compiler
-AC_PATH_PROG(MCS, mcs, no)
-AC_PATH_PROG(GMCS, gmcs, no)
-AC_PATH_PROG(DMCS, dmcs, no)
+AC_PATH_PROG(MCS, mono-csc, no)
+AC_PATH_PROG(GMCS, mono-csc, no)
+AC_PATH_PROG(DMCS, mono-csc, no)
 AC_PATH_PROG(RUNTIME, mono, no)
 AC_PATH_PROG(GACUTIL, gacutil, no)
 
Index: mono-tools/Mono.Profiler/compile
===================================================================
--- mono-tools.orig/Mono.Profiler/compile
+++ mono-tools/Mono.Profiler/compile
@@ -1 +1 @@
-gmcs -debug -out:profiler-decoder.exe profiler-decoder-library/*.cs profiler-file-decoder/*.cs
+mono-csc -debug -out:profiler-decoder.exe profiler-decoder-library/*.cs profiler-file-decoder/*.cs
