--- mono-tools-2.0.orig/configure.in
+++ mono-tools-2.0/configure.in
@@ -62,7 +62,7 @@
 
 if test "x$enable_gecko" = "xyes"; then
    # get the mozilla home directory
-   MOZILLA_HOME="`$PKG_CONFIG --variable=libdir mozilla-gtkmozembed`"
+   MOZILLA_HOME="`$PKG_CONFIG --variable=libdir xulrunner-gtkmozembed`"
    AC_SUBST(MOZILLA_HOME)
 fi
 
--- mono-tools-2.0.orig/ilcontrast/ilcontrast.in
+++ mono-tools-2.0/ilcontrast/ilcontrast.in
@@ -19,16 +19,20 @@
 elif test x@MOZILLA_HOME@ != x; then
     if [ -f @MOZILLA_HOME@/chrome/comm.jar ]; then
         MOZILLA_HOME=@MOZILLA_HOME@
+        MOZILLA_FIVE_HOME=$MOZILLA_HOME
     fi 
 elif grep GRE_PATH /etc/gre.d/*.conf > /dev/null ; then
         MOZILLA_HOME=$(grep -h GRE_PATH= /etc/gre.d/*.conf | cut -d '"' -f 2 -d = | head -n 1)
+        MOZILLA_FIVE_HOME=$MOZILLA_HOME
 elif [ $(which xulrunner 2> /dev/null) ] > /dev/null ; then
     MOZILLA_FIVE_HOME=`getdirectory xulrunner`
     MOZILLA_HOME=$MOZILLA_FIVE_HOME
 elif [ $(which mozilla 2> /dev/null) ] && grep MOZILLA_FIVE_HOME= "$(which mozilla)" > /dev/null ; then
 	MOZILLA_HOME=$(grep MOZILLA_FIVE_HOME= $(which mozilla) | cut -d '"' -f 2)
+	MOZILLA_FIVE_HOME=$MOZILLA_HOME
 elif [ $(which firefox 2> /dev/null) ] && grep MOZILLA_FIVE_HOME= "$(which firefox)" > /dev/null ; then
 	MOZILLA_HOME=$(grep MOZILLA_FIVE_HOME= $(which firefox) | cut -d '"' -f 2)
+	MOZILLA_FIVE_HOME=$MOZILLA_HOME
 fi
 
 if [ -n $LD_LIBRARY_PATH ]; then
