--- mono-tools-2.0.orig/gendarme/rules/Gendarme.Rules.Portability/Makefile.am
+++ mono-tools-2.0/gendarme/rules/Gendarme.Rules.Portability/Makefile.am
@@ -13,7 +13,7 @@
 CLEANFILES = $(rules_dll) $(rules_dll).mdb $(tests_dll)
 DISTCLEANFILES = Makefile.in Gendarme.Rules.Portability.xml TestResult.xml
 
-EXTRA_RULES_OPTIONS = -r:ICSharpCode.SharpZipLib.dll -pkg:dotnet
+EXTRA_RULES_OPTIONS = -r:ICSharpCode.SharpZipLib.dll -r:System.Web.Services
 
 rules_sources = \
 	DoNotHardcodePathsRule.cs \
