--- mono-tools-2.0.orig/configure.in
+++ mono-tools-2.0/configure.in
@@ -119,13 +119,13 @@
 dnl
 dnl I hate PKG_CONFIG_MODULES, that is for the weak
 dnl
-if pkg-config --modversion cecil>/dev/null; then
-AC_MSG_RESULT("Found cecil.pc")
+if pkg-config --modversion mono-cecil>/dev/null; then
+AC_MSG_RESULT("Found mono-cecil.pc")
 else
-AC_MSG_ERROR([You are missing Cecil (cecil.pc for packageconfig) required for gui-compare])
+AC_MSG_ERROR([You are missing Cecil (mono-cecil.pc for packageconfig) required for gui-compare])
 fi
 
-CECIL_ASM=`pkg-config --variable=Libraries cecil`
+CECIL_ASM=`pkg-config --variable=Libraries mono-cecil`
 AC_SUBST(CECIL_ASM)
 
 dnl Intl
